utilities/scan_logs.py

```diff
@@ -72,6 +72,27 @@ def scan_for_errors(log_file_path, error_strings):
                     raise LogScanningError('{} found!! Please review: {} '.format(word, log_file_path))
 
 
+def scan_for_pattern(logfile_path, pattern_list):
+    """
+    Scans a log file line by line for a provided array of words.
+
+    'pattern_list' should be a list. Example ['panic', 'error', 'data race']
+    """
+    if type(pattern_list) != list:
+        raise ValueError('error_strings must be a list')
+
+    # Scan each line in the log file for the words to search for
+    flag = False
+    with open(logfile_path) as f:
+        for line in f:
```

Comment:
you can use "for line in open(log file_path)". It optimise the code

Reply:
done